clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10120,9 +10120,12 @@ def err_conflicting_sycl_kernel_attributes : Error<
 def err_sycl_attibute_cannot_be_applied_here
     : Error<"%0 attribute cannot be applied to a "
             "%select{static function or function in an anonymous namespace"
-            "|class member function"
-            "|function with a raw pointer return type"
-            "|function with a raw pointer parameter type}1">;
+            "|class member function}1">;
+def warn_sycl_attibute_address_space_ext
+    : Warning<"SYCL 1.2.1 specification does not allow %0 attribute applied "
+              "to a %select{function with a raw pointer return type"
+              "|function with a raw pointer parameter type}1">,
```

Comment:
I think you can move "function with a raw pointer " outside of the selector.

Reply:
Done.